query/btccheckpoint.go

```diff
@@ -1,58 +1,72 @@
 package query
 
 import (
+	"context"
+
 	btcctypes "github.com/babylonchain/babylon/x/btccheckpoint/types"
-	"github.com/babylonchain/rpc-client/client"
+	"github.com/cosmos/cosmos-sdk/client"
 	sdkquerytypes "github.com/cosmos/cosmos-sdk/types/query"
 )
 
-// BTCCheckpointParams queries btccheckpoint module's parameters via ChainClient
-func BTCCheckpointParams(c *client.Client) (*btcctypes.Params, error) {
-	ctx, cancel := c.GetDefaultQueryContext()
+// QueryBTCCheckpoint queries the BTCCheckpoint module of the Babylon node
+// according to the given function
+func (c *QueryClient) QueryBTCCheckpoint(f func(ctx context.Context, queryClient btcctypes.QueryClient)) {
```

Comment:
We can make the function parameters return an error if something went wrong. This way, the function parameters can have more control in terms of the errors that they have and what they return (e.g. a function parameter might have two different kinds of errors that it returns, and we want the caller of the wrapper to return one of those).

Reply:
That's a good point! fixed